src/Caching/Impl/ModuleUniqueId.cs

```diff
@@ -70,23 +91,37 @@ public static string GetUniqueId(this IPythonModule module, IServiceContainer se
                     if (folders.Length == 1) {
                         var fileName = Path.GetFileNameWithoutExtension(folders[0]);
                         var dash = fileName.IndexOf('-');
-                        return $"{moduleName}({fileName.Substring(dash + 1)})";
+                        id = $"{moduleName}({fileName.Substring(dash + 1)})";
+                        break;
                     }
                     // Move up if nothing is found.
                     versionFolder = Path.GetDirectoryName(versionFolder);
                 }
             }
 
-            var config = interpreter.Configuration;
-            if (moduleType == ModuleType.CompiledBuiltin || string.IsNullOrEmpty(filePath) || modulePathType == PythonLibraryPathType.StdLib) {
-                // If module is a standard library, unique id is its name + interpreter version.
-                return $"{moduleName}({config.Version.Major}.{config.Version.Minor})";
+            if (id == null) {
+                var config = interpreter.Configuration;
+                if (moduleType == ModuleType.CompiledBuiltin || string.IsNullOrEmpty(filePath) || modulePathType == PythonLibraryPathType.StdLib) {
+                    // If module is a standard library, unique id is its name + interpreter version.
+                    id = $"{moduleName}({config.Version.Major}.{config.Version.Minor})";
+                }
+            }
+
+            if (id == null) {
+                var parent = moduleResolution.CurrentPathResolver.GetModuleParentFromModuleName(moduleName);
+                if (parent == null) {
+                    id = moduleName;
```

Comment:
Won't this string just be something like "foo.bar" without any versioning? If the version needs to come from a parent, I'd expect to have seen recursion or something. Maybe I'm just reading this wrong.

Reply:
This is an edge case just to handle `GetModuleParentFromModuleName` that can return `null` since hashing needs parent `IImportChildrenSource`.